src/rt/lifetime.d

```diff
@@ -406,6 +406,22 @@ size_t __arrayPad(size_t size, const TypeInfo tinext) nothrow pure @trusted
     return size > MAXMEDSIZE ? LARGEPAD : ((size > MAXSMALLSIZE ? MEDPAD : SMALLPAD) + structTypeInfoSize(tinext));
 }
 
+/**
+  clear padding that might not be zeroed by the GC (it assumes it is within the
+  requested size from the start, but it is actually at the end of the allocated block)
+  */
+void __arrayClearPad(ref BlkInfo info, size_t arrsize, size_t padsize) nothrow pure
```

Comment:
`private` ?

Reply:
done.

Reply:
On the wrong function?

Reply:
Oops, fixed. Just shows how arbitrary it looks to mark only this function private.